src/components/updatemenus/attributes.js

```diff
@@ -49,6 +49,7 @@ var buttonsAttrs = {
 
 module.exports = {
     _isLinkedToArray: 'updatemenu',
+    _arrayAttrRegexps: [/^updatemenus\[(0|[1-9][0-9]+)\]\.buttons/],
```

Comment:
that wouldn't work for the other instance of this, [`/^mapbox([2-9]|[1-9][0-9]+)?\.layers/`](https://github.com/plotly/plotly.js/pull/1403/files#diff-26aaae812b2edd7de7c8d4ab79df84f0R19)